x-pack/plugins/ingest_pipelines/public/application/components/pipeline_editor/components/processor_form/processors/user_agent.tsx

```diff
@@ -78,7 +95,39 @@ export const UserAgent: FunctionComponent = () => {
           'xpack.ingestPipelines.pipelineEditor.userAgentForm.propertiesFieldHelpText',
           { defaultMessage: 'Properties added to the target field.' }
         )}
-        propertyOptions={propertyOptions}
+        euiFieldProps={{
+          options: propertyOptions as [],
```

Comment:
I think if you update the interface for `propertyOptions` to `Array<EuiComboBoxOptionOption<string>>`, you won't need to typecast here.